src/app/qgisapp.cpp

```diff
@@ -16046,6 +16046,55 @@ void QgisApp::triggerCrashHandler()
 #endif
 }
 
+void QgisApp::addTabifiedDockWidget( Qt::DockWidgetArea area, QDockWidget *dockWidget, const QStringList &onTopOf )
+{
+  QList< QDockWidget * > dockWidgetsInArea;
+  const auto dockWidgets = findChildren< QDockWidget * >();
+  for ( QDockWidget *w : dockWidgets )
+  {
+    if ( w->isVisible() && dockWidgetArea( w ) == area )
+    {
+      dockWidgetsInArea << w;
+    }
+  }
+
+  addDockWidget( area, dockWidget );  // First add the dock widget, then attempt to tabify
+  if ( dockWidgetsInArea.length() > 0 )
+  {
+    QDockWidget *onTopOfDockWidget = nullptr;
+    if ( !onTopOf.isEmpty() )
+    {
+      QStringListIterator iter( onTopOf );
+      while ( iter.hasNext() )
+      {
+        const QString &objectName = iter.next();
+        for ( QDockWidget *cw : dockWidgetsInArea )
+        {
+          if ( cw->objectName() == objectName )
+          {
+            onTopOfDockWidget = cw;
+            iter.toBack();  // The task is done, exit the iterator
+            break;
+          }
+        }
+      }
+    }
+
+    if ( !onTopOfDockWidget )
+    {
+      onTopOfDockWidget = dockWidgetsInArea.at( 0 );  // Last resort
+    }
+    tabifyDockWidget( onTopOfDockWidget, dockWidget );
+
+    // Select our newly added tab
+    QgsDockWidget *qdw = dynamic_cast< QgsDockWidget * >( dockWidget );
```

Comment:
I still think we should leave this to be a manual step done by the caller when needed. I don't think we always want to force raise newly added docks.

Reply:
What about a boolean parameter `raiseTab` that is `true` by default?

Reply:
I'm just not sure how often we want to encourage developers to raise by default. I'm concerned that this could be (unintentionally) abused by plugins  -- if they call the method without setting raise to false, then on startup a user will constantly see their dock tab order reset with the plugin's docks moving to front...

Reply:
Well, it could be `false` by default to avoid that. 
I would just like to let this method be in charge of raising the tab.

If you still think it's not its business, let me know. I'll implement it as it better fits with QGIS as a whole.

Reply:
I'm ok if you make it false by default!

Reply:
Implemented in https://github.com/qgis/QGIS/pull/35933/commits/3c9a375173a1f529d3af1cdc54d741502ab6366c